packages/flutter_tools/lib/src/build_system/tools/asset_transformer.dart

```diff
@@ -0,0 +1,148 @@
+    await asset.copy(tempInputFile.path);
+    File tempOutputFile = _fileSystem.systemTempDirectory.childFile(getTempFilePath());
+
+    try {
+      for (final (int i, AssetTransformerEntry transformer) in transformerEntries.indexed) {
+        final _AssetTransformerFailure? transformerFailure = await _applyTransformer(
+          asset: tempInputFile,
+          output: tempOutputFile,
+          transformer: transformer,
+          workingDirectory: workingDirectory,
+        );
+
+        if (transformerFailure != null) {
+          _logger.printError(
+            'User-defined transformation of asset "${asset.path}" failed.\n'
+            '${transformerFailure.message}',
+          );
+          return false;
+        }
+
+        ErrorHandlingFileSystem.deleteIfExists(tempInputFile);
+        if (i == transformerEntries.length - 1) {
+          await tempOutputFile.copy(outputPath);
+        } else {
+          tempInputFile = tempOutputFile;
+          tempOutputFile = _fileSystem.systemTempDirectory.childFile(getTempFilePath());
+        }
+      }
+    } finally {
+      ErrorHandlingFileSystem.deleteIfExists(tempInputFile);
+      ErrorHandlingFileSystem.deleteIfExists(tempOutputFile);
+    }
+
+    return true;
+  }
+
+  Future<_AssetTransformerFailure?> _applyTransformer({
+    required File asset,
+    required File output,
+    required AssetTransformerEntry transformer,
+    required String workingDirectory,
+  }) async {
+    final List<String> transformerArguments = <String>[
+      '--input=${asset.absolute.path}',
+      '--output=${output.absolute.path}',
+      ...?transformer.args,
+    ];
+
+    final List<String> command = <String>[
+      _dartBinaryPath,
+      'run',
+      transformer.package,
+      ...transformerArguments,
+    ];
+
+    final ProcessResult result = await _processManager.run(
+      command,
+      workingDirectory: workingDirectory,
+    );
+    final String stdout = result.stdout as String;
+    final String stderr = result.stderr as String;
+
+    if (result.exitCode != 0) {
+      return _AssetTransformerFailure(
+        'Transformer process terminated with non-zero exit code: ${result.exitCode}\n'
+        'Transformer package: ${transformer.package}\n'
+        'Full command: ${command.join(' ')}\n'
+        'stdout:\n$stdout\n'
+        'stderr:\n$stderr'
+      );
+    }
+
+    if (!_fileSystem.file(output).existsSync()) {
+      return _AssetTransformerFailure(
+        'Asset transformer ${transformer.package} did not produce an output file.\n'
+        'Input file provided to transformer: "${asset.path}"\n'
+        'Expected output file at: "${output.absolute.path}"\n'
+        'Full command: ${command.join(' ')}\n'
+        'stdout:\n$stdout\n'
+        'stderr:\n$stderr',
+      );
+    }
+
+    return null;
+  }
+}
+
+final class _AssetTransformerFailure {
+  _AssetTransformerFailure(this.message);
```

Comment:
uber-nit `const`. I know it's pointless today, but some day someone will make this `@visibleForTesting` and hard code a const string :)